packages/waku/tests/create-pages.test.ts

```diff
@@ -73,6 +73,14 @@ describe('type tests', () => {
         staticPaths: ['x', 'y', 'z'],
         component: () => 'Hello',
       });
+
+      // also good
+      createPage({
+        render: 'static',
+        path: '/test/[a]/[b]',
```

Comment:
@dai-shi This was commented as being an invalid slug, but from my understanding of the docs, nested segment routes are supported in static mode? https://waku.gg/#nested-segment-routes

It seems like what is not supported is the "catch-all" wildcard slugs, so I updated the tests and types here to reflect the docs as best I could.

Reply:
Thanks!!

Reply:
Is my understanding correct though? There is also code to test that static wildcard support does exist:

https://github.com/dai-shi/waku/blob/main/packages/waku/tests/create-pages.test.ts#L518-L527

https://github.com/dai-shi/waku/blob/main/packages/waku/src/router/create-pages.ts#L196-L201

I can change these in this PR as well of course... I just wanted to be careful and call this code out because it seems to be in conflict with the docs at the moment.

Reply:
I guess so. But, I'm not super confident with anything. I would not be surprised if there were differences among implementation, tests and docs.
I'm fine if you completely re-implement it from scratch around this path handling (as it's not tightly related with `unstable_defineRouter` which I'll be modifying.)

Reply:
Sounds good to me.

Here's what I'll plan to start with: 

Breaking up the 4 different types of pages into 4 different functions for handling

They can all be defined inside createPages to share other logic and utilities, but a good starting point to me would be unraveling this code a bit to start making things more clear.

I'll update more here as I make progress and start a discussion if we need to make larger design decisions.

Reply:
Following back up here: I updated the tests and disabled wildcard/catch-all routes for  for the static render mode fully.

My guess is that these were added initially in error when adding all types of slug support for each render mode.

But when I stop to think about "what would it mean to be a wildcard route in a static site?" there really is no way to support this.

For a wildcard route to work, you would need to run logic on the request to match the request to the wildcard slug and then inject that as a value into the component for that page. This is simply not possible when you are requesting a static file from a static file server. Your request is taken as a literal representation for the html file you are interested in.

Finally, I am confident this changeset is ready for a look and I will start a discussion with some follow up thoughts about the `createPages` code 

I'll leave this thread open in case there are some further thoughts around this :)

Reply:
resolved this by appropriately supporting wildcards for static mode... some examples:

```ts
createPage({
  render: 'static',
  path: '/nested/[...ids]',
  // results in paths /nested/foo/bar and /nested/foo/baz and /nested/foo
  staticPaths: [['foo', 'bar'], ['foo','baz'], ['foo']],
  component: ...,
});  
```